### PR TITLE
test: ups test coverage and light refactoring

### PR DESCRIPTION
**Pull request by bsbodden**
* d7c7afc - test: test EnableXXXRepositories class discovery methods
* 6520586 - refactor: remove commented out code
* ffae43c - refactor: pull test config static class for reuse
* d746a4c - test: test that persistent entities must not be null
* 7135ccf - test: Hashes cannot persist collection with nulls
* 4efd72e - test: add test for byte[] serialization in Hashes
* 8860c66 - refactor: remove unnecessary empty checks in Hash binary converters
* df4d6e9 - refactor: remove unreachable null check
* 7f9c034 - test: add missing test for empty tuple
* 6353162 - test: tests metamodel generator
* 9e9a7ba - build: add commented out showWarning element to debug annotation processor
* e356d58 - refactor: extract searchIndexFieldAliasFor method from index creation code
* 3e31a35 - test: add test for pageable catchall FT search

### CODE REVIEW

File: redis-om-spring/pom.xml [hunk 1]
```diff
@@ -242,6 +252,7 @@
           <source>${maven.compiler.source}</source>
           <target>${maven.compiler.target}</target>
           <encoding>${project.build.sourceEncoding}</encoding>
+          <!-- showWarnings>true</showWarnings -->
```
Comment (slorello89) [inline redis-om-spring/pom.xml:255]: this should probably be removed

File: redis-om-spring/src/main/java/com/redis/om/spring/convert/BytesToLocalDateConverter.java [hunk 2]
```diff
@@ -7,17 +7,12 @@
...
 @ReadingConverter
 public class BytesToLocalDateConverter implements Converter<byte[], LocalDate> {
 
   @Override
   public LocalDate convert(byte[] source) {
-    if (ObjectUtils.isEmpty(source)) {
-      return null;
-    }
-
     return LocalDate.ofInstant(Instant.ofEpochSecond(Long.parseLong(toString(source))), ZoneId.systemDefault());
```
Comment (slorello89) [inline redis-om-spring/src/main/java/com/redis/om/spring/convert/BytesToLocalDateConverter.java:16]: if null is still possible, shouldn't this return an Option<T>?

File: redis-om-spring/src/main/java/com/redis/om/spring/convert/BytesToLocalDateTimeConverter.java [hunk 3]
```diff
@@ -7,17 +7,12 @@
...
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.data.convert.ReadingConverter;
-import org.springframework.util.ObjectUtils;
 
 @ReadingConverter
 public class BytesToLocalDateTimeConverter implements Converter<byte[], LocalDateTime> {
 
   @Override
   public LocalDateTime convert(byte[] source) {
-    if (ObjectUtils.isEmpty(source)) {
```
Comment (slorello89) [inline redis-om-spring/src/main/java/com/redis/om/spring/convert/BytesToLocalDateTimeConverter.java:17]: same comment above

File: redis-om-spring/src/main/java/com/redis/om/spring/search/stream/actions/BaseAbstractAction.java [hunk 4]
```diff
@@ -14,11 +14,7 @@ public abstract class BaseAbstractAction implements TakesJSONOperations {
   public BaseAbstractAction(Field field) {
     this.field = field;
     Optional<Field> maybeId = ObjectUtils.getIdFieldForEntityClass(field.getDeclaringClass());
-    if (maybeId.isPresent()) {
-      this.idField = maybeId.get();
-    } else {
-      throw new IllegalArgumentException(field.getDeclaringClass().getName() + " does not appear to have an ID field");
-    }
+    this.idField = maybeId.get();
```
Comment (slorello89) [inline redis-om-spring/src/main/java/com/redis/om/spring/search/stream/actions/BaseAbstractAction.java:17]: if we're removign the check - do we really need this to be an Optional?

**Review by slorello89 — CHANGES_REQUESTED** [review]

File: redis-om-spring/src/main/java/com/redis/om/spring/convert/BytesToLocalDateConverter.java (hunk 2 above)
Comment (bsbodden) [reply redis-om-spring/src/main/java/com/redis/om/spring/convert/BytesToLocalDateConverter.java:16]: No, after  checking the class that manipulates these (in Spring Data Redis) it only uses them with a non-null byte source

File: redis-om-spring/src/main/java/com/redis/om/spring/search/stream/actions/BaseAbstractAction.java (hunk 4 above)
Comment (bsbodden) [reply redis-om-spring/src/main/java/com/redis/om/spring/search/stream/actions/BaseAbstractAction.java:17]: The check was unnecessary, classes without an Id do not get this far! The annotation processor fails before they get here.

**Review by slorello89 — APPROVED** [review]
lgtm 👍
